addons/project/models/project.py

```diff
@@ -2720,22 +2720,32 @@ def _get_default_color(self):
         ('name_uniq', 'unique (name)', "A tag with the same name already exists."),
     ]
 
-    def _get_project_tags_domain(self, domain, project_id):
-        tag_ids = list(self.with_user(SUPERUSER_ID)._search(
-            ['|', ('task_ids.project_id', '=', project_id), ('project_ids', 'in', project_id)]))
-        return expression.AND([domain, [('id', 'in', tag_ids)]])
-
     @api.model
     def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
         if 'project_id' in self.env.context:
-            domain = self._get_project_tags_domain(domain, self.env.context.get('project_id'))
+            tag_ids = self._name_search()
+            domain = expression.AND([domain, [('id', 'in', tag_ids)]])
         return super().read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
 
     @api.model
     def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
-        if 'project_id' in self.env.context:
-            domain = self._get_project_tags_domain(domain, self.env.context.get('project_id'))
-        return super().search_read(domain=domain, fields=fields, offset=offset, limit=limit, order=order)
+        tag_ids = self._name_search()
+        domain = expression.AND([domain, [('id', 'in', tag_ids)]])
+        return self.sort_tag_records_by_id(super().search_read(domain=domain, fields=fields, offset=offset, limit=limit, order=order), tag_ids)
```

Comment:
order is overridden by the sort_tag_records_by_id. when should we take it into account?